samples/client/petstore/go/go-petstore/enum_class.go

```diff
@@ -10,5 +10,12 @@
 
 package petstore
 
-type EnumClass struct {
-}
+type EnumClass string
+
+// List of EnumClasss
+const (
+	_abc EnumClass = "_abc"
```

Comment:
There are several functions you will need to override in the Go generator. Please have a look at https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PhpClientCodegen.java#L648-L706 for reference.